lib/segment/src/data_types/named_vectors.rs

```diff
@@ -74,35 +113,63 @@ impl<'a> NamedVectors<'a> {
         self.map.iter().map(|(k, _)| k.as_ref())
     }
 
+    // TODO(sparse) remove this function. There is only one usage and this fn is logically strange
     pub fn into_default_vector(mut self) -> Option<Vec<VectorElementType>> {
-        self.map
-            .get_mut(DEFAULT_VECTOR_NAME)
-            .map(|src| std::mem::take(src).into_owned())
+        self.map.get_mut(DEFAULT_VECTOR_NAME).map(|src| match src {
+            CowValue::Dense(v) => std::mem::take(v).into_owned(),
+            CowValue::Sparse(_v) => unreachable!(),
+        })
     }
 
-    pub fn into_owned_map(self) -> HashMap<String, Vec<VectorElementType>> {
+    pub fn into_owned_map(self) -> HashMap<String, Vector> {
         self.map
             .into_iter()
-            .map(|(k, v)| (k.into_owned(), v.into_owned()))
+            .map(|(k, v)| {
+                (
+                    k.into_owned(),
+                    match v {
+                        CowValue::Dense(src) => Vector::Dense(src.into_owned()),
```

Comment:
can we use the new `to_owned` implementation for `CowValue` there?

Reply:
Fixed!